go/vt/discovery/healthcheck.go

```diff
@@ -448,8 +448,16 @@ func (hc *HealthCheckImpl) updateHealth(th *TabletHealth, prevTarget *query.Targ
 			}
 		}
 	case isPrimary && !isPrimaryUp:
-		// No healthy master tablet
-		hc.healthy[targetKey] = []*TabletHealth{}
+		if healthy, ok := hc.healthy[targetKey]; ok && len(healthy) > 0 {
+			var newHealthy []*TabletHealth
```

Comment:
i don't know if this is the proper way to do the deletion in this case, please let me know if you have better suggestions

Reply:
This is fine too.